core/unix/os_public.h

```diff
@@ -242,4 +242,10 @@ typedef kernel_sigcontext_t sigcontext_t;
 #    define SC_RETURN_REG SC_A0
 #endif /* X86/ARM */
 
+#if defined(MACOS) && defined(AARCH64)
```

Comment:
Add an `XXX i#5383: ` comment explaining the error you observed when adding this attribute on x86 (like in your PR).